magda-web-client/src/Components/Dataset/Add/DatasetAddCommon.ts

```diff
@@ -933,6 +946,7 @@ async function convertStateToDatasetRecord(
             "dataset-distributions": {
                 distributions: distributionRecords.map((d) => d.id)
             },
+            "ckan-export": ckanExport,
```

Comment:
Here we save the whole `ckan-export` aspect using record API may erase the `ckanId` (which is used by minion to identified ckan package created by minion) that is generated from backend (as our record API will not merge the object and only replace the whole ckan-export object). 
As mentioned in slack, we probably want to use patch APIs (https://dev.magda.io/api/v0/apidocs/index.html#api-Registry_Record_Aspects-PatchV0RegistryRecordsRecordidAspectsAspectid or http://192.168.64.3:30100/api/v0/apidocs/index.html#api-Registry_Record_Service-PatchV0RegistryRecordsId) to handle export aspect saving separately.
i.e. Using a separate patch API call to specifically ask registry API to update status fields (and `exportRequired` field) of export aspect without touching other data fields (e.g. `ckanId`) might be generated from the backend.
We probably want to implement the logic in the function  `submitDatasetFromState` (still in DatasetAddCommon.ts) as it will only be called when the dataset is published.

Reply:
Created #2887 to come back to this. At the moment it fetches the `ckanExport` aspect again to stay current